Documentation/proposals/202309-controller-id.md

```diff
@@ -0,0 +1,65 @@
+# Controller-ID Proposal
+
+* Owners:
+  * [danielmellado](https://github.com/danielmellado)
+* Related Tickets:
+  * [#4281](https://github.com/prometheus-operator/prometheus-operator/issues/4281)
+  * [#4498](https://github.com/prometheus-operator/prometheus-operator/pull/4498)
+* Other Docs:
+  * n/a
+
+# Introduction
+
+This proposal aims to implement a solution to support multiple cluster-level
+Prometheus instances running concurrently without conflicting over the same
+custom resources.
+
+This issue can significantly impact use cases where multiple Prometheus operator
+instances run in parallel.
+
+# Why
+
+Currently, we encounter issues when different users deploy different instances
+of the Prometheus operator, that will try to reconcile the same resources.
+
+In the worst-case scenario, these operators may not only compete for ownership
+of the CRD resources but also attempt to rewrite or redeploy different versions
+of the CRD, causing disruptions to all pods.
+
+The remediation for this scenario, where users deploy their Prometheus operator
+instances in parallel, involves using one of the many CLI arguments, such as
+`--deny-namespaces`, `--namespaces`, `--prometheus-instance-selector`, or
+`prometheus-instance-namespaces`.
+
+# How
+
+After some research, we have identified a potential solution already
+implemented by the
+[zalando/postgres-operator](https://github.com/zalando/postgres-operator). When
+an operator is configured with a specific "controller ID" value, it will only
+reconcile resources that have a matching "controller ID" annotation.
+
+Conversely, if the operator is not configured with a "controller ID," it will
+skip all resources that have a "controller ID" annotation. More details can be
+found in the
+[zalando/postgres-operator documentation](https://github.com/zalando/postgres-operator/blob/master/docs/administrator.md#operators-with-defined-ownership-of-certain-postgres-clusters).
+
+# Goals
+
+* Guarantee that a custom resource will be managed by a specific Prometheus
+  operator instance.
+
```

Comment:
For non-goals, I'd add
*  it's not going to be an zero-configuration solution (e.g. it still requires work from the user deploying the operator + resource).
* this will be an opt-in feature

Reply:
You mean "This will not be an opt-in feature"?

Reply:
I assume that Simon means that users would be able to not trigger the feature? Not sure why they wouldn't like to but just in case they would opt-out the config?

Reply:
I mean to say that when you start the operator without any specific argument, it will attempt to reconcile all resources in all namespaces. But I realize that it's similar to the first non-goal.

Reply:
I'll remove the 2nd and put a clarification on the first one.